src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +29,15 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        String[] addressArr = address.split(",");
+        if (addressArr.length!=4) {
+        	throw new IllegalValueException(MESSAGE_ADDRESS_ERROR);
```

Comment:
Consider holding the literal 4 in a named constant. Otherwise someone new to your code won't understand the significance of the value 4 and why you are checking against it.